### PR TITLE
Enable python tests

### LINKED ISSUE #4: Setup Python CI in Github Action
# Problem 

Setup Python Github Action to test python codebase.

### PR DESCRIPTION
**Pull request by eddyxu**
Enable python tests with

* Ubuntu 22.04 X Python 3.8-3.10
* macOS 12 + Python 3.10

Closes #4

### CODE REVIEW

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        brew update
+        brew install apache-arrow protobuf
```
Comment (changhiskhan) [inline .github/workflows/python.yml:50]: don't need to install parquet separately?

Comment (changhiskhan) [reply .github/workflows/python.yml:50]: and also arrow-python arrow-dataset?

File: .github/workflows/python.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - name: Build C++
+      working-directory: cpp
+      run: |
+        cmake -B build
+        make -C build -j
+    - name: Test Pip install
+      working-directory: python
+      run: |
+        python -m pip install -q wheel numpy pyarrow cython
```
Comment (changhiskhan) [inline .github/workflows/python.yml:59]: should we use `python -m build` so if we pin any version in the build deps of pyproject.toml it's respected?

File: python/tools/Dockerfile.manylinux2014 [hunk 3]
```diff
@@ -7,10 +7,6 @@ RUN yum update -y \
...
     && yum install -y --enablerepo=epel arrow-devel \
         arrow-dataset-devel arrow-python-devel parquet-devel wget \
-    && cd /tmp \
-    && wget -q https://github.com/protocolbuffers/protobuf/releases/download/v3.20.1/protobuf-cpp-3.20.1.tar.gz -O - | tar -xz \
-    && cd protobuf* \
-    && ./configure \
-    && make -j \
-    && make install \
-    && rm -rf /tmp/proto*
+        ftp://ftp.pbone.net/mirror/ftp.centos.org/7.9.2009/cloud/x86_64/openstack-train/Packages/p/protobuf-devel-3.6.1-4.el7.x86_64.rpm \
```
Comment (changhiskhan) [inline python/tools/Dockerfile.manylinux2014:10]: does this need any epel / release system parsing magic?

**Review by changhiskhan — APPROVED** [review]
Some nits

File: .github/workflows/python.yml (hunk 1 above)
Comment (eddyxu) [reply .github/workflows/python.yml:50]: on mac, `apache-arrow` installs all of them, this is a mega-package.

File: .github/workflows/python.yml (hunk 2 above)
Comment (eddyxu) [reply .github/workflows/python.yml]: ok, gonna just use `ciwheelbuilder` which reads `pyproject.toml` to get these packages.

File: python/tools/Dockerfile.manylinux2014 (hunk 3 above)
Comment (eddyxu) [reply python/tools/Dockerfile.manylinux2014:10]: i could not find the package for this one.

**Pull request merged**